docs(oauth): updates for custom config of oauth

PaulRMellor opened the pull request:
**Documentation**

Updates chapter 17 of the Deploying guide related to enabling OAuth 2.0 token-based access.
Aligns the content with the new custom authentication type.

The chapter now concentrates on Strimzi-specific configuration, deferring to the strimzi-kafka-oauth project and Keycloak documentation for additional background information. If there are Keycloak-specific or other details that are essential for users, we can add them back within the new structure.

**Key changes:**

- Focus on Strimzi config setup
  - Configuring OAuth on Kafka listeners
  - Configuring Strimzi components and external clients
  - Enabling authorization on brokers
 - Migration guidance from the deprecated oauth and keycloak types.
- Streamlines explanations and defers OAuth-specific content to the Strimzi OAuth project. (oauth project readme being expanded to cover advanced configuration options)
- Updates all examples to use the custom type.

Note: SASL/PLAIN is not covered, but guidance can be added if required.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

scholzj commented:
@mstruk Any chance you can have a look at this?

mstruk commented on documentation/modules/oauth/con-oauth-server-config.adoc:
```diff
@@ -0,0 +1,160 @@
+* The listener uses the `OAUTHBEARER` SASL mechanism.
+* Tokens are validated by calling the OAuth 2.0 introspection endpoint for each new client connection.
+* The broker authenticates to the protected introspection endpoint using a client ID and a client secret mounted into the container.
+* The JAAS login module and callback handler perform token validation and map a specified attribute value from introspection endpoint response to a Kafka principal.
```
We can make it more explicit:
```suggestion
* The JAAS login module and callback handler perform token validation and map `preferred_username` attribute value from introspection endpoint response to a Kafka principal.
```

mstruk commented on documentation/modules/oauth/con-oauth-server-config.adoc:
```diff
@@ -0,0 +1,160 @@
+        volumes:
+          - name: oauth-credentials
+            secret:
+              name: kafka-broker-oauth-secret
```
I guess this volume mounts the `/opt/oauth/credentials/client-secret` file. The way to use that approach would be to map the content of this file to the value of the environment variable `OAUTH_CLIENT_SECRET`. Something like:
```
env:
  - name: OAUTH_CLIENT_SECRET
    valueFrom:
      secretKeyRef:
        name: kafka-broker-auth-secret
        key: secret
```

PaulRMellor replied:
I've removed the volume config and added a note on loading configuration values from external sources

mstruk commented on documentation/modules/oauth/con-oauth-client-config.adoc:
```diff
@@ -0,0 +1,176 @@
+        org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required
+          oauth.token.endpoint.uri="https://<auth_server_address>/<path_to_token_endpoint>"
+          oauth.client.id="kafka-bridge"
+          oauth.client.secret="${strimzienv:OAUTH_CLIENT_SECRET}";
```
Does this actually work? Can this be used and is automatically expanded into the env var value when KafkaBridge is started? If so, the same approach can be used when configuring the `oauth.client.secret` in the listener if the same kind of expansion takes place.

PaulRMellor replied:
It was based on the example in the proposal.
we have some docs on: [loading external values](https://strimzi.io/docs/operators/in-development/deploying#assembly-loading-config-with-providers-str)

PaulRMellor commented:
Thanks for the comments @mstruk 
All addressed. I've updated the server-side config to remove the volumes and added a note on loading external values.
With the client, I've left as it is. I don't know if we want to show one way or the other for both.
